pytradfri/device/base_controller.py

```diff
@@ -16,9 +23,13 @@ def __init__(self, device: Device) -> None:
         self._device = device
 
     @property
-    def raw(self):
+    @abstractmethod
+    def raw(
+        self,
+    ) -> list[BlindResponse] | list[SocketResponse] | list[
```

Comment:
Please extract the id attribute of the sub device response models to a parent model class that we define here in the base controller. Then inherit from this new parent class in all the sub device response models. Then set the new parent model response type as the list item return type here.

The base controller class shouldn't know any details of the sub device classes, only the common parts.

Reply:
```
pytradfri/device/socket_control.py:14: error: Return type "List[SocketResponse]" of "raw" incompatible with return type "List[BaseResponse]" in supertype "BaseController"  [override]
```

Do we need to `cast`? It sounds by response that we shouldn't need to.

Reply:
Try using `Sequence` instead of `list` as container type.

https://docs.python.org/3/library/collections.abc.html#collections.abc.Sequence

Reply:
That seems to have worked! 💯